### PR TITLE
[ADD][10.0] website_snippet_data_slider

### PR DESCRIPTION
**Pull request by lasley**
Bringing this over from the LasLabs repos for v10 conversion, I figure OCA might like 😉 
- Create new module to provide a SlickJS data slider snippet for website

This can be functionally reviewed by using the "DataSlider" link on the website. Note that you have to be logged in for the partner implementation due to model permissions.

You can obviously also just drag the widget wherever you want, although you'll need to install `product` in order to not edit the options in the HTML - default implementation is on `product.template`. It's not a dependency though, because the slider can be used on anything else - such as partner in the demo.

I'm all ears if you have a good way to allow text input in widget editing, instead of requiring editing of the HTML. The JS events only seem to like selections ☹️

### CODE REVIEW

File: website_snippet_data_slider/README.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* Consolidate the data options a bit?
+* Find a way to use slug, instead of URI Prefix
+* Touch up the stylesheets
+* Add a real thumbnail, instead of icon
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (yajo) [inline website_snippet_data_slider/README.rst]: oops

File: website_snippet_data_slider/demo/data_slider_demo.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                                                    &quot;data_image_field&quot;: &quot;image_medium&quot;,
+                                                    &quot;data_name_field&quot;: &quot;name&quot;,
+                                                    &quot;data_title&quot;: &quot;Partners&quot;,
+                                                    &quot;data_title_tag&quot;: &quot;h1&quot;,
+                                                    &quot;slidesToShow&quot;: 5,
+                                                    &quot;slidesToScroll&quot;: 2,
+                                                    &quot;dots&quot;: true,
+                                                    &quot;infinite&quot;: true,
+                                                    &quot;arrows&quot;: true,
+                                                    &quot;autoplay&quot;: true
```
Comment (yajo) [inline website_snippet_data_slider/demo/data_slider_demo.xml]: I'd prefer it if you wrap this attribute between single quotes and use normal `"` if possible, simply for readability.

File: website_snippet_data_slider/demo/data_slider_demo.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                                 }"
+                                 >
+                            <div class="container">
+                                <div class="row">
+                                    <div class="oe_structure" />
```
Comment (yajo) [inline website_snippet_data_slider/demo/data_slider_demo.xml]: The demo snippet should be inside this element to allow demo users to modify it.

File: website_snippet_data_slider/static/src/css/data_slider.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/* Copyright (C) 2016 LasLabs, Inc. [https://laslabs.com]
+ * @author Dave Lasley <dave@laslabs.com>
+ * @license LGPL-3
```
Comment (yajo) [inline website_snippet_data_slider/static/src/css/data_slider.css]: Always add the license link ;)

File: website_snippet_data_slider/static/src/js/data_slider.editor.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+ */
+
+odoo.define('website_snippet_data_slider.snippet_editor', function(require){
+  "use strict";
+  
+  var options = require('web_editor.snippets.options');
+  
+  options.registry.data_slider = options.Class.extend({
+    
+    defaults: require('website_snippet_data_slider').defaults,
```
Comment (yajo) [inline website_snippet_data_slider/static/src/js/data_slider.editor.js]: The normal thing is to declare requires at the start of module.

File: website_snippet_data_slider/static/src/js/data_slider.editor.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    
+    },
+    
+    set_size: function() {
+      var size = this.get_size();
+      var $slickEl = this.$target.find('.o_slick_container');
+      this.$target.height(size.height);
+      $slickEl.css({
+          'position' : 'relative',
+          //'left': ((size.width - $slickEl.width()) / 2),
```
Comment (yajo) [inline website_snippet_data_slider/static/src/js/data_slider.editor.js]: :hankey:

File: website_snippet_data_slider/views/snippet_template.xml [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                            class="slick-option">
+                            <a>Domain</a>
+                        </li>
+                        <li t-att-data-settings_callback='"{\"%s\": \"%s\" }" % ("data_image_field", true)'
+                            class="slick-option">
+                            <a>Image Field</a>
+                        </li>
+                    </ul>
+                </li>
+                -->
```
Comment (yajo) [inline website_snippet_data_slider/views/snippet_template.xml]: :hankey: :question:

**Review by yajo — CHANGES_REQUESTED** [review]
The snippet is quite cool, it deserves a good icon, and a screenshot in the description.

captura de pantalla de 2016-10-26 10-20-51

**Comment by Shiraj-Nisus** [comment]
I have noticed some JavaScript issue with this new version.Tested this with Odoo V10 Ent, Firefox 49.0.2, Chrome 54.0.2840.71 m and Edge.
As soon as I made changes to DataSlider by selecting options, browser were crashed.It is not responding at all.Tested this 3 times and same result.See the attached image for more details.
dataslider

File: website_snippet_data_slider/demo/data_slider_demo.xml (hunk 2 above)
Comment (lasley) [reply website_snippet_data_slider/demo/data_slider_demo.xml]: Problem with single quotes is that they don't allow multi-line. I was stuck between a rock and a hard place of having a gargantuan line or using the `&quot;`. I'll switch to one line though, sure

**Comment by lasley** [comment]
Thanks for the reviews.

@Yajo - all comments attended to

@3sPoojanath - I was able to replicate the issue, then I couldn't after I made the other changes, so I'm hoping that did it. Would you mind confirming in EE? I only have CE to test in unfortunately.

**Comment by 3sPoojanath** [comment]
@lasley - It is still the same and I have tested in EE.
script

**Review by yajo — APPROVED** [review]

**Comment by Shiraj-Nisus** [comment]
@lasley - Tested on EE with fresh db and confirmed that issue was fixed!

**Comment by lasley** [comment]
We good for merge on this? IMO @3sPoojanath provided an implicit thumb with testing and subsequent confirmation of fix.

I'd like to begin my backport of this version for v9 so I can fully offload this app from LasLabs repos

**Comment by pedrobaeza** [comment]
I have tried the module, but I don't see how to add slides manually.

**Comment by lasley** [comment]
The slides are added via the domain of the query & have no manual selection. 

Unfortunately the Odoo widget control panel only allows for A/B/C option selections, so the more fancy attributes like domain have to be edited through [HTML editor](https://github.com/OCA/website/pull/271/files#diff-7f6a91b39c76d644bf136068d87f5081R17).

**Comment by lasley** [comment]
I feel like I now need to update the ReadMe now, because if it was unclear to Pedro it's definitely going to be unclear to others. I'll try to think of some better phrasing

**Comment by Shiraj-Nisus** [comment]
@lasley Yes I confirmed it worked perfectly on Enterprise version.By the way I noticed this new version also creating a partner slider automatically with a menu.Both worked perfectly on EE.

**Comment by lasley** [comment]
Thanks @3sPoojanath 

Yeah there is demo data in this version to assist with functional reviews. That menu and slider should not show up unless you have demo data enabled on the Odoo instance 😄 . I used partner so that the `website_sale` dependency for product slider isn't required.

**Comment by pedrobaeza** [comment]
@lasley, sorry, it's my fault. I didn't read properly the README and I misunderstood the goal of the snippet: I thought that the snippet was a container that provides slides that allows you to drag and drop other snippets as slides.

**Comment by lasley** [comment]
Whoops I forgot I was doing something here. I clarified the Readme, pretty sure there's no mistaking the intent now 😄

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**